test/distributed/test_device_mesh.py

```diff
@@ -61,6 +61,58 @@ def test_init_process_group(self):
         self.assertTrue(is_initialized())
         self.destroy_pg()
 
+    @with_comms
+    def test_reuse_process_group(self):
+        from torch.distributed import new_group
+
+        tp_group_0 = new_group([0, 1])
+        tp_group_1 = new_group([2, 3])
+        dp_group_0 = new_group([0, 2])
+        dp_group_1 = new_group([1, 3])
+
+        # test init_device_mesh with mesh_dim_names
+        mesh_shape = (2, self.world_size // 2)
+        mesh_dim_names = ("DP", "TP")
+        mesh_2d = init_device_mesh(
+            self.device_type, mesh_shape, mesh_dim_names=mesh_dim_names
+        )
+
+        from torch.distributed.distributed_c10d import _world
+
+        tags_to_pg = _world.tags_to_pg
+
+        # For each rank, there should be 4 tags_to_pg, including tags:
+        #   1) default tag for user PGs (which contains all the pgs on a particular rank)
+        #   2) tag for world size pg
+        #   3) tag for tp pg
+        #   4) tag for dp pg
+        self.assertEqual(len(tags_to_pg), 4)
```

Comment:
can we also check before/after `init_device_mesh`, `tags_to_pg` does not change?

Reply:
> can we also check before/after `init_device_mesh`, `tags_to_pg` does not change?

Good idea. Will add that to test.

Reply:
> can we also check before/after `init_device_mesh`, `tags_to_pg` does not change?

Actually, we cannot directly compare `tags_to_pg` before/after `init_device_mesh` call, since PG cannot be pickled. I am using `tags_to_pg_name` for comparison instead to make sure they are the same before/after.